src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3414,6 +3414,37 @@ GenTree* Lowering::LowerHWIntrinsicCndSel(GenTreeHWIntrinsic* node)
             blendVariableId = NI_EVEX_BlendVariableMask;
             op1             = maskNode;
         }
+        else if (op2->IsVectorZero() || op3->IsVectorZero())
+        {
+            // If either of the value operands is const zero, we can optimize down to AND or AND_NOT.
+            GenTree* binOp = nullptr;
+
+            if (op3->IsVectorZero())
```

Comment:
can we have nodes of type `ConvertMaskToVector(Vector.Zero)` for `op2` or `op3`. I just opened https://github.com/dotnet/runtime/pull/114272 to fix a case where I was not checking that for arm64.

Reply:
I don't see a way we'd create that currently on xarch, but I plan on reviving #110342 and will keep that possibility in mind since we'll more eagerly choose intrinsics that produce a mask with that.